### ISSUE TITLE
React handles paths on client side, but server does not

### ISSUE THREAD

**Issue by jakubgs** [dap-ps/discover#9]
## Description

Because React apps are contained all within one JS file which is loaded by `index.html` by default the Express.js server does not handle any other paths than what is present in the `frontend` folder.

## Reproduction

1. Go to https://dev.dap.ps and click "View All" just above the Categories.
2. This will take you to https://dev.dap.ps/discover-dapps/all
3. However, if you try and go directly to https://dev.dap.ps/discover-dapps/all it falls over.

## Details

This is important because of the way the Status browser handles history. It stores the full URL, so if you go back (using top left back button) to Home Screen, and then try navigate back into, say, one of the categories, it falls over.

## Links

* https://stackoverflow.com/questions/27928372/react-router-urls-dont-work-when-refreshing-or-writing-manually

**Comment by jakubgs** [comment]
The way this works right now is that the Express.js server simply hosts what's inside the `frondend` folder which is generated by the build process:
https://github.com/dap-ps/discover/blob/19608fadc5c96a9c0f745cab2cac2a21faf7b356/back-end/server.js#L25
Which means only files that are present there are hosted by the server:
```bash
 $ ls -l full-build/frontend 
total 32
-rw-r--r-- 1 sochan sochan 2608 Aug  8 09:50 asset-manifest.json
-rw-r--r-- 1 sochan sochan 1182 Aug  8 09:50 favicon.ico
drwxr-xr-x 2 sochan sochan 4096 Aug  8 09:50 fonts
drwxr-xr-x 3 sochan sochan 4096 Aug  8 09:50 images
-rw-r--r-- 1 sochan sochan 2094 Aug  8 09:50 index.html
-rw-r--r-- 1 sochan sochan 3110 Aug  8 09:50 precache-manifest.4dda4d6c020aa3f36670ed2c9d2dd286.js
-rw-r--r-- 1 sochan sochan 1182 Aug  8 09:50 service-worker.js
drwxr-xr-x 5 sochan sochan 4096 Aug  8 09:50 static
```
So anything that isn't there returns `404`:
```bash
 $ curl -si https://dev.dap.ps/discover-dapps/all | head -n1 
HTTP/1.1 404 Not Found
```

**Comment by jakubgs** [comment]
Here's a proposed fix: #10
Simply return `index.html` for any path that doesn't match any file we have.

**Comment by jakubgs** [comment]
Deployed:
```
 $ curl -si https://dev.dap.ps/discover-dapps/all | head -n5
HTTP/1.1 200 OK
Accept-Ranges: bytes
Access-Control-Allow-Origin: *
Cache-Control: public, max-age=0
Content-Type: text/html; charset=UTF-8
```
